enshrine-esdt-safe/src/from_sovereign/transfer_tokens.rs

```diff
@@ -36,13 +39,70 @@ pub trait TransferTokensModule:
             sc_panic!("Operation is not registered");
         }
 
+        let are_tokens_registered =
+            self.verify_operation_tokens_issue_paid(operation.tokens.clone());
+
+        if !are_tokens_registered {
+            self.emit_transfer_failed_events(
+                &hash_of_hashes,
+                &OperationTuple {
+                    op_hash: operation_hash.clone(),
+                    operation: operation.clone(),
+                },
+            );
+
+            return;
+        }
+
         let minted_operation_tokens = self.mint_tokens(&operation.tokens);
         let operation_tuple = OperationTuple {
-            op_hash: operation_hash,
-            operation,
+            op_hash: operation_hash.clone(),
+            operation: operation.clone(),
         };
 
-        self.distribute_payments(hash_of_hashes, operation_tuple, minted_operation_tokens);
+        self.distribute_payments(
+            hash_of_hashes.clone(),
+            operation_tuple,
+            minted_operation_tokens,
+        );
+    }
+
+    #[endpoint(registerNewTokenID)]
+    #[payable("*")]
+    fn register_new_token_id(&self, tokens: MultiValueEncoded<TokenIdentifier>) {
+        let call_payment = self.call_value().single_esdt().clone();
+        let wegld_identifier = self.wegld_identifier().get();
+
+        require!(
+            call_payment.token_identifier == wegld_identifier,
+            "WEGLD is the only token accepted as register fee"
+        );
+
+        require!(
+            call_payment.amount == DEFAULT_ISSUE_COST * tokens.len() as u64,
+            "WEGLD fee amount is not met"
+        );
+
+        for token_id in tokens {
+            self.register_token(token_id);
```

Comment:
If this makes token issue, it won't work. You can only make one legacy async call per call, which is what the token issue framework uses.

Reply:
`register_token` just saves the `TokenIdentifier` to the `paid_issued_tokens` mapper